#! /usr/bin/env python

PACKAGE='realsense_id_ros'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name                      Type Level Description                                                                                       Default       Min    Max
gen.add("camera_rotation",        int_t,  0, "Enable the algorithm to work with a rotated device.",                                                  0,        0,   180)
gen.add("security_level",         str_t,  0, "Set security level to 'high' to allow no mask suport or to level 'medium' to support masks.",                    "medium")
gen.add("algo_flow",              str_t,  0, "Algorithms which will be used during authentication: 'all', 'detection', 'recognition' and 'spoof'.",               "all")
gen.add("face_selection_policy",  str_t,  0, "Face selection policy to run authentication on 'all' (up to 5) detected faces vs 'single' (closest) face.",         "all")


exit(gen.generate(PACKAGE, "realsense_id_ros", "RealSenseIDParameters"))

